c++/test/tcompound.cpp

```diff
@@ -98,7 +98,9 @@ test_compound_2()
         bkg  = static_cast<unsigned char *>(malloc(nelmts * sizeof(dst_typ_t)));
         orig = static_cast<unsigned char *>(malloc(nelmts * sizeof(src_typ_t)));
         for (i = 0; i < nelmts; i++) {
-            s_ptr       = (reinterpret_cast<src_typ_t *>(orig)) + i;
+            H5_WARN_CAST_ALIGNMENT_OFF
+            s_ptr = (reinterpret_cast<src_typ_t *>(orig)) + i;
```

Comment:
If these few places are the only locations where `cast-align` is currently suppressed, I'd much rather these be fixed (maybe in a separate PR) than continuing to be suppressed. The solution is very simple so no real need to keep suppressing `cast-align`.